server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -864,6 +873,20 @@ private SchemaInfo normalizeSchemaBinary(SchemaInfo schemaInfo) {
         return new SchemaInfo(schemaInfo.getType(), schemaInfo.getSerializationFormat(), schemaBinary, schemaInfo.getProperties());
     }
 
+    private void validateJsonSchema(String schemaString) {
+        try {
+            // 1. try draft 3
+            // jackson JsonSchema only supports json draft 3. If the schema definition is not compatible with draft 3, 
+            // try parsing the schema with everit library which supports drafts 4 6 and 7. 
+            OBJECT_MAPPER.readValue(schemaString, JsonSchema.class);
+        } catch (IOException e) {
+            JSONObject rawSchema = new JSONObject(new JSONTokener(schemaString));
+            // draft 4 to 7
+            SchemaLoader.builder().useDefaults(true).draftV7Support().schemaJson(rawSchema)
```

Comment:
Would this work for all versions above draft 3?

Also, what if there are any exceptions thrown at this point? I suppose RuntimeException might be getting thrown and should be caught here.

Reply:
RuntimeException is caught by the caller which declares the schema validation a failure.